color.cpp

```diff
@@ -214,6 +215,33 @@ void test_color()
   }
 #endif // FIX_ISSUE_460
 
+#ifdef FIX_ISSUE_628
+  // (628) A random color between red, blue, green can be sampled
+  {
+    color c;
+    int n_red = 0;
+    int n_green = 0;
+    int n_blue = 0;
+
+    const color r = create_red_color();
+    const color g = create_green_color();
+    const color b = create_blue_color();
+
+    std::mt19937 rng(1);
+    for (int i = 0; i < 100; ++i)
+      {
+        c = get_random_rgb(rng);
+        if (c == r) { ++n_red; }
+        else if (c == g) { ++n_green; }
+        else if (c == b) { ++n_blue; }
+        else { throw("get_random_rgb should not return any other color than r, g, b"); }
+      }
+
+    assert(n_red > 0 && n_red < 100);

```

Comment:
I see the idea here; it would be weird if `n_red` would be more than 100. However, that would mean we are testing the test itself. I feel that is overkill.

Hence I propose to use `assert(n_red > 0);`, as this resembles test best ('At least once, we should get red'). To add more weight behind this argument, adding the second part results in a needless fail when a user increases the number of for loop iterations to e.g. 1000.

Reply:
I meant to check that not all iterations (that is, less than 100) result in red. But indeed, this is redundant with the next lines and adds needless complexity!